server/permit.js

```diff
@@ -0,0 +1,21 @@
+// middleware for doing role-based permissions
+//expects allowed user roles as params and returns a middleware which will either
+//pass to the next middleware or send a 403 based on the user role
+const permit = (...allowed) => {
+  const isAllowed = role => allowed.indexOf(role) > -1
+
+  // return a middleware
+  return (req, res, next) => {
+    let userRole
+    if (req.user.isAdmin) userRole = 'Admin'
```

Comment:
right now req.user is undefined (we're never setting it for an unauthenticated user) so this line of code errors out on unauthenticated (500 error instead of expected 403 error)